### PR TITLE
Codelist/variable subclass

### PR DESCRIPTION
**Pull request by luciecastella**
Linked to #137 

Create a VariableCodeList subclass

### CODE REVIEW

**Review by phackstock — CHANGES_REQUESTED** [review]
Very solid start @luciecastella.
A few comments below. 
One more thing to take care of is to go into the `tests/` folder and look for any reference to `CodeList` that should now be `VariableCodeList`. E.g. https://github.com/luciecastella/nomenclature/blob/codelist/variable-subclass/tests/test_codelist.py#L13.
There might be some region specific changes we made to the `mapping` attribute of `CodeList` so if you want you can try and reduce the complexity of the type hint. Since we're going to change it so that CodeLists contain Code instances anyway this is probably overkill at this point.

File: nomenclature/definition.py [hunk 1]
```diff
@@ -35,7 +35,10 @@ def __init__(self, path, dimensions=["region", "variable"]):
 
         self.dimensions = dimensions
         for dim in self.dimensions:
-            self.__setattr__(dim, CodeList.from_directory(dim, path / dim))
+            if dim == "variable":
```
[the remark this answers is not in the document]
Comment (luciecastella) [reply nomenclature/definition.py:38]: (implementing it now so that I don't forget)

**Review by phackstock — APPROVED** [review]
Very nice, looks good from my side. 
There's a couple of things that will be cleaned up in the next few PRs but for this one I'd say we're good.

**Review by danielhuppmann — APPROVED** [review]
Looks nice, thanks @luciecastella!

**Pull request merged**